server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -132,9 +133,14 @@
     public interface Custom extends NamedDiffable<Custom>, ToXContentFragment {
 
         EnumSet<XContentContext> context();
-    }
 
-    public interface NonRestorableCustom extends Custom {}
+        /**
+         * @return true if this custom could be restored from snapshot
+         */
+        default boolean isRestorable() {
+            return context().contains(XContentContext.SNAPSHOT);
```

Comment:
I think this is fine, though I would like to hear the opinion of @original-brownbear.

Reply:
Provided the context values are all correct (this was a bit of an issue when me and Ievgen last checked if I remember correctly) this should be fine. I couldn't find any way this would not work since the context isn't something we read from the repo but just have per class => all good from my end.